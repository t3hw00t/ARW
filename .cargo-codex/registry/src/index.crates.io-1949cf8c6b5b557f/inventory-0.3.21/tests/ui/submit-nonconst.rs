struct Thing;

impl Thing {
    fn new() -> Self {
        Thing
    }
}

inventory::collect!(Thing);

inventory::submit!(Thing::new());

fn main() {}
