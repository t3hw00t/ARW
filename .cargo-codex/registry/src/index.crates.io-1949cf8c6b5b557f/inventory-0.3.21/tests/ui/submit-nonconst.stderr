error[E0015]: cannot call non-const associated function `Thing::new` in statics
  --> tests/ui/submit-nonconst.rs:11:20
   |
11 | inventory::submit!(Thing::new());
   |                    ^^^^^^^^^^^^
   |
   = note: calls in statics are limited to constant functions, tuple structs and tuple variants
   = note: consider wrapping this expression in `std::sync::LazyLock::new(|| ...)`
